Python Examples/TestMonitor/test_data_manager_client.py

```diff
@@ -0,0 +1,182 @@
+    :return: json response after creating the results
+    """
+    if len(results) == 0 :
+        raise ValueError("Number of results needs to be created can not be empty")
+    body = create_test_result_request(results)
+    request_uri = base_uri + create_results_route
+    request_response = raise_post_request(request_uri, body)
+
+    return request_response.json()
+
+def update_results(results: List) -> Dict:
+    """
+    Updates existing test results by merging or replacing values.
+    :param results: Results which needs to be updated
+    :return: json response after updating the results
+    """
+    if len(results) == 0 :
+        raise ValueError("Number of results needs to be updated can not be empty")
+    body = update_test_results_request(results, determine_status_from_steps=True)
+    request_uri = base_uri + update_results_route
+    request_response = raise_post_request(request_uri, body)
+
+    return request_response.json()
+
+def create_steps(steps: List) -> Dict:
+    """
+    Creates new test steps from the supplied models. The result associated with the step must exist prior to step creation. The server automatically generates step ids if not supplied.
+    :param steps: Steps which needs to be created
+    :return: json response after creating steps
+    """
+    if len(steps) == 0 :
+        raise ValueError("Number of steps needs to be created can not be empty")
+    body = test_step_create_or_update_request_object(
+            steps, update_result_total_time=True
+        )
+    request_uri = base_uri + create_steps_route
+    request_response = raise_post_request(request_uri, body)
+
+    return request_response.json()
+
+def update_steps(steps: List) -> Dict:
+    """
+    Updates existing steps by merging or replacing values.
+    :param steps: Steps which needs to be updated
+    :return: json response after updating the steps
+    """
+    if len(steps) == 0 :
+        raise ValueError("Number of steps needs to be updated can not be empty")
+    body = test_step_create_or_update_request_object(
+                steps, update_result_total_time=True
+            )
+    request_uri = base_uri + update_steps_route
+    request_response = raise_post_request(request_uri, body)
+
+    return request_response.json()
+
+def delete_result(result_id: str, delete_steps: bool = True) -> None:
+    """
+    Deletes the existing test result.
+    :param result_id: result id which needs to be deleted
+    """
+    if result_id == None :
+        raise ValueError("Missing required parameter 'result_id' when calling ResultsApi->DeleteResultV2")
+    request_uri = f"{base_uri}{delete_result_route}/{result_id}?deleteSteps{delete_steps}"
+    raise_delete_request(request_uri)
+
+def delete_results(result_ids: List, delete_steps: bool = True) -> Dict:
+    """
+    Deletes the existing test results.
+    :param result_ids: result ids which needs to be deleted
+    """
+    if len(result_ids) == 0 or result_ids == None:
```

Comment:
If `result_ids` is `None`, then checking its length will throw an exception, so we have to check for nullability first

Reply:
updated to check for the nullability first and then for the length.